torchao/_models/llama/generate.py

```diff
@@ -268,6 +269,15 @@ def main(
 
         if "fp6" in quantization:
             quantize_(model, fpx_weight_only(3, 2))
+        if "uintx" in quantization:
+            # uintx-nbits-group_size
+            # "uintx-2-64"
+            _quant_args = quantization.split("-")
+            nbits = int(_quant_args[1])
+            _NBITS_TO_DTYPE = {1: torch.uint1, 2: torch.uint2, 3: torch.uint3, 4: torch.uint4, 5: torch.uint5, 6: torch.uint6, 7: torch.uint7, 8: torch.uint8}
```

Comment:
nit can you add checks in here to for the valid options, see the groupsize checks in int4 for example

Reply:
sounds good